Nomad: document fallback behavior for Consul tokens (#2637)
hashicorp/web-unified-docs: content/nomad/v1.10.x/content/docs/secure/acl/consul.mdx

## Nomad Workload Identities

> Nomad clients can use a task or service's [Workload Identity][nomad_wid] to

Reviewer: ```suggestion
Nomad clients can use a task or service's [workload identity][nomad_wid] to
```

authenticate to Consul and obtain an ACL token specific to the service or
task. When using Nomad workload identities, every allocation gets its own Consul
ACL token(s).

> Without workload identity, allocations fallback to using the Nomad client

Reviewer: ```suggestion
Without workload identity, allocations fall back to using the Nomad client
```

agent's own Consul token. This is a simpler approach but allows all workloads on
the same node access to the same data from Consul. If this is suitable for your
environment, refer to [Consul Without Workload Identity][].

By default, Nomad does not generate workload identities for services, and tasks

> only receive an identity that can be used to access data from Nomad itself, such

Reviewer: ```suggestion
only receive an identity that they can use to access data from Nomad itself, such
```
style nit: use active voice

module can help you automate the process of applying configuration to a Consul
cluster.

## Consul Without Workload Identity

Without workload identity, Nomad clients use their own Consul ACL token to
register services, read services and KV for `template` blocks, and configure
Consul service mesh.

For service registration and Consul service mesh, the client automatically falls
back to use its own Consul token for any workload that doesn't have a workload

> identity for Consul from the server. This will include any allocation that was

Reviewer: ```suggestion
identity for Consul from the server. This includes any allocation that was
```

created before the servers were configured to use workload identity.

> For `template` blocks you'll need additional configuration to allow job template

Reviewer: ```suggestion
For `template` blocks you need additional configuration to allow job template
```

blocks to use the Nomad client's token, which may have broader capabilities than
a task strictly needs. Set [`client.template.use_client_consul_token =
true`][use-client-consul-token] in the Nomad client's agent configuration to
allow the template runner access to the token. Add the following block to the
Consul policy used by the Nomad client to allow the client to read Consul KV
data in `template` blocks.

```hcl
key_prefix "" {
  policy = "read"
}
```

## Migrating to Using Workload Identity with Consul

Nomad version 1.7.0 introduced Workload Identity authentication, and deprecated
authenticating via a token passed to the `job run` command. Token authentication
was then removed in Nomad 1.10.0. Migrating from the legacy (pre-1.7) workflow
where workload use the agent's Consul token requires configuation on your Consul
cluster and your Nomad server agents. It does not require updating your running

> Nomad jobs. To migrate:

Reviewer: ```suggestion
authenticating with a token passed to the `job run` command. We then removed token authentication
in Nomad 1.10.0. Migrating from the legacy (pre-1.7) workflow
where workloads use the agent's Consul token requires configuation on your Consul
cluster and your Nomad server agents. It does not require updating your running
Nomad jobs. To migrate:
```
